src/App.js

```diff
@@ -13,10 +14,14 @@ export default class App extends Component {
     this.state = {
       path,
       params,
-      cards: JSON.parse(cards) || []
+      cards: JSON.parse(cards) || [],
+      practiceIndex: 0,
```

Comment:
You should let `Practice` manage its own state. Just pass the cards as a prop and let it take over from there.